### PR TITLE
Add ser/de of Typing packet

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/toxcore/messenger/packet/typing.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+impl FromBytes for TypingStatus {
+    named!(from_bytes<TypingStatus>,
+        switch!(le_u8,
+            0 => value!(TypingStatus::NotTyping) |
+            1 => value!(TypingStatus::Typing)
+        )
+    );
+}
+
+/** Typing is a struct that holds typing status of me.
```
Comment (kpp) [inline src/toxcore/messenger/packet/typing.rs]: of me?

File: src/toxcore/messenger/packet/typing.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*! Typing struct.
+*/
+
+use nom::le_u8;
+
+use crate::toxcore::binary_io::*;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[repr(u8)]
```
Comment (kurnevsky) [inline src/toxcore/messenger/packet/typing.rs]: What is it for?

File: src/toxcore/messenger/packet/typing.rs (hunk 1 above)
Comment (kpp) [reply src/toxcore/messenger/packet/typing.rs]: Plz fix

File: src/toxcore/messenger/packet/typing.rs (hunk 2 above)
Comment (kurnevsky) [reply src/toxcore/messenger/packet/typing.rs]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f61bd5f33844aecbf719174b567f1458

Comment (kpp) [reply src/toxcore/messenger/packet/typing.rs]: wow

Comment (NamsooCho) [reply src/toxcore/messenger/packet/typing.rs]: done

File: src/toxcore/messenger/packet/typing.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Length    | Content
+--------- | ------
+`1`       | `0x33`
+`1`       | Typing status(0 = not typing, 1 = typing)
+
+*/
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Typing {
+    status: TypingStatus,
```
Comment (kurnevsky) [inline src/toxcore/messenger/packet/typing.rs]: How about unifying with https://github.com/tox-rs/tox/pull/316/files#diff-50f8bfc82d815d2951af7e519042f913R29 - you used short form of struct there, but full here.

Comment (NamsooCho) [reply src/toxcore/messenger/packet/typing.rs]: done

Comment (kurnevsky) [reply src/toxcore/messenger/packet/typing.rs]: And the same question for UserStatus then :)

**Pull request merged**